modules/dnn/src/model.cpp

```diff
@@ -637,6 +637,335 @@ void DetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+                offsetX = (Width - size.width * resizeFactor) / 2.0;
+                offsetY = (Height - size.height * resizeFactor) / 2.0;
+
+                // Adjust the box coordinates
+                box.x = (box.x - offsetX) / resizeFactor;
+                box.y = (box.y - offsetY) / resizeFactor;
+                box.width = (box.width - offsetX) / resizeFactor;
+                box.height = (box.height - offsetY) / resizeFactor;
+            } else {
+                // Handle other cases or throw an error if the padding mode is unsupported
+                CV_Error(Error::StsNotImplemented, "Unsupported padding mode");
+            }
+        }
+    }
+};
+
+YOLODetectionModel::YOLODetectionModel(const String& model, const String& config, const int version)
+{
+    impl = makePtr<YOLODetectionModel_Impl>(version);
+    impl->initNet(readNet(model, config));
+    impl.dynamicCast<YOLODetectionModel_Impl>()->setNetworkType(true);
+}
+
+YOLODetectionModel::YOLODetectionModel(const String& onnx, const int version)
+{
+    impl = makePtr<YOLODetectionModel_Impl>(version);
+    impl->initNet(readNetFromONNX(onnx));
+}
+
+YOLODetectionModel::YOLODetectionModel()
+{
+    impl = std::static_pointer_cast<Model::Impl>(makePtr<YOLODetectionModel_Impl>());
+}
+
+void YOLODetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+                            CV_OUT std::vector<float>& confidences, CV_OUT std::vector<Rect2d>& boxes,
+                            float confThreshold, float nmsThreshold){
+
+    CV_Assert(impl != nullptr && impl.dynamicCast<YOLODetectionModel_Impl>() != nullptr); // remove once default constructor is removed
+
+    std::vector<Mat> detections;
+    impl.dynamicCast<YOLODetectionModel_Impl>()->processFrame(frame, detections);
+
+    postProccess(
+        detections,
+        boxes,
+        confidences,
+        classIds,
+        confThreshold,
+        nmsThreshold,
+        impl.dynamicCast<YOLODetectionModel_Impl>()->getYoloVersion(),
+        impl.dynamicCast<YOLODetectionModel_Impl>()->getNetworkType(),
+        getNmsAcrossClasses(),
+        impl->size.width,
+        impl->size.height);
+
+    impl.dynamicCast<YOLODetectionModel_Impl>()->boxRescale(boxes);
+}
+
+void YOLODetectionModel::postProccess(
+    std::vector<Mat>& detections,
+    CV_OUT std::vector<Rect2d>& keep_boxes,
+    CV_OUT std::vector<float>& keep_confidences,
+    CV_OUT std::vector<int>& keep_classIds,
+    const float confThreshold,
+    const float nmsThreshold,
+    const int yoloVersion,
+    const bool darknet,
+    const bool nmsAcrossClasses,
+    int frameWidth,
+    int frameHeight
+    ){
+
+    // Retrieve
+    std::vector<int> classIds;
+    std::vector<float> confidences;
+    std::vector<Rect2d> boxes;
+
+    // NOTE: here the number 8 (see below) refers to yolov8
+    if (yoloVersion == 8){
+        cv::transposeND(detections[0], {0, 2, 1}, detections[0]);
+    }
```

Comment:
I still think that we can determine YOLOv8 from others just by shape, not an input argument

Reply:
done